modele-social/règles/entreprise-établissement.yaml

```diff
@@ -110,25 +110,151 @@ entreprise . imposition . IS . notification IS non intégré:
-          plafond: 500000 €/an
-        - taux: 33.3%
+      variations:
+        - si: exercice . début >= 01/01/2022
+          alors:
+            tranches:
+              - taux: 15%
+                plafond: 38120 €/an * prorata taux réduits
+              - taux: 25%
+        - si: exercice . début >= 01/01/2021
+          alors:
+            tranches:
+              - taux: 15%
+                plafond: 38120 €/an * prorata taux réduits
+              - taux: 26.5%
+        - si: exercice . début >= 01/01/2020
+          alors:
+            tranches:
+              - taux: 15%
+                plafond: 38120 €/an * prorata taux réduits
+              - taux: 28%
+        - si: exercice . début >= 01/01/2019
+          alors:
+            tranches:
+              - taux: 15%
+                plafond: 38120 €/an * prorata taux réduits
+              - taux: 28%
+                plafond: 500000 €/an * prorata taux réduits
+              - taux: 31%
+        - si: exercice . début >= 01/01/2018
+          alors:
+            tranches:
+              - taux: 15%
+                plafond: 38120 €/an * prorata taux réduits
+              - taux: 28%
+                plafond: 500000 €/an * prorata taux réduits
+              - taux: 33.3333%
+    arrondi: oui
+  références:
+    Fiche impots.gouv.fr: https://www.impots.gouv.fr/portail/international-professionnel/impot-sur-les-societes
+    Fiche service-public.fr: https://www.service-public.fr/professionnels-entreprises/vosdroits/F23575
+
+entreprise . impôt sur les sociétés . prorata taux réduits:
+  unité: '%'
+  formule:
+    variations: 
+      - si: éligible taux réduits
+        alors: prorata durée de l'exercice
+      - sinon: 0 €/an
+  note: |
+    Publicodes ne nous permet pas de définir qu'une tranche "taux réduit" d'un
+    barème est non applicable, nous utilisons donc à la place une multiplication
+    par zéro pour implémenter la non éligibilité aux taux réduits.
+
+entreprise . impôt sur les sociétés . éligible taux réduits:
+  formule:
+    toutes ces conditions:
+      - chiffre d'affaires <= 7630 k€/an * prorata durée de l'exercice
+      - nom: capital détenu au moins à 75 pourcents par des personnes physiques
+        valeur: oui
+
+
+entreprise . impôt sur les sociétés . prorata durée de l'exercice:
+  description: |
+    Lorsque la durée de l’exercice n'est pas égale à un an, on pro-ratise les
+    plafonds utilisés dans le barème de l'impôt sur les sociétés.
+  titre: pro-ratisation
+  unité: '%'
+  formule: exercice . durée / 1 an
+  # Note: c'est un peu plus subtil que cela : « En cas d’exercice ouvert ou
+  # arrêté en cours de mois calendaire, le nombre de jours résiduels concourt à
+  # la détermination du rapport pour un montant égal au rapport existant entre
+  # ce nombre et 30. »
+  références:
+    Bofip: https://bofip.impots.gouv.fr/bofip/2065-PGP.html/identifiant%3DBOI-IS-LIQ-20-20-20180801
+    
+entreprise . impôt sur les sociétés . contribution sociale:
```

Comment:
Je serai d'avis de l'enlever si on ne l'utilise pas...

Reply:
C'est vrai que ça concerne pas vraiment les TPE (2,5M de benef) mais j'aime bien l'idée d'avoir une implémentation complète. Je pense l'exposer dans le simulateur avec une ligne en dessous de l'IS.

Mais du coup ça pose aussi la question de l'éligibilité aux taux réduit (qui dépend du CA) et donc de rajouter un composant `<Question />` dans ce cas là.